File: api/src/main/java/ro/moment/api/domain/Message.java
package ro.moment.api.domain;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class Message extends BaseEntity{

    @OneToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @OneToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "group_id", referencedColumnName = "id")
    private Group group;

    @ManyToOne()
    @JoinColumn(name = "parent_message_id", referencedColumnName = "id")
    private Message parentMessage;

    private String title;
    private String content;
    private LocalDateTime publishDate;
}

File: ui/1Moment/src/app/groups/group-discussion-page/Message.ts
export interface Message{
  id :number,
  userId: number,
  groupId: number,
  parentMessageId: string,
  title?: string,
  content: string,
  publishDate: Date,
  username?: string,
  groupName?: string,
  liked?: boolean,
  likeCount?: number,
}

